code/modules/mob/living/simple_animal/hostile/mining_mobs/elites/elite.dm

```diff
@@ -0,0 +1,354 @@
+	if(boosted)
+		mychild.maxHealth = mychild.maxHealth * 2
+		mychild.health = mychild.maxHealth
+	return
+		
+/obj/structure/elite_tumor/Initialize()
+	. = ..()
+	AddComponent(/datum/component/gps, "Menacing Signal")
+	START_PROCESSING(SSobj, src)
+	
+/obj/structure/elite_tumor/Destroy()
+	STOP_PROCESSING(SSobj, src)
+	mychild = null
+	return ..()
+	
+/obj/structure/elite_tumor/process()
+	if(isturf(loc))
+		for(var/mob/living/simple_animal/hostile/asteroid/elite/elitehere in loc)
+			if(elitehere == mychild && activity == TUMOR_PASSIVE)
+				mychild.adjustHealth(-mychild.maxHealth*0.05)
+				var/obj/effect/temp_visual/heal/H = new /obj/effect/temp_visual/heal(get_turf(mychild))
+				H.color = "#FF0000"
+				mychild.update_health_hud()
+		
+/obj/structure/elite_tumor/attackby(obj/item/W, mob/user, params)
+	. = ..()
+	if(istype(W, /obj/item/organ/regenerative_core) && activity == TUMOR_INACTIVE && !boosted)
+		var/obj/item/organ/regenerative_core/core = W
+		if(core.preserved)
+			visible_message("<span class='boldwarning'>As [user] drops the core into the tumor, the tumor appears to swell.</span>")
+			icon_state = "advanced_tumor"
+			boosted = TRUE
+			light_range = 6
+			desc = "[desc]  This one seems to glow with a strong intensity."
+			qdel(core)
+			return TRUE
+			
+/obj/structure/elite_tumor/proc/arena_trap()
+	var/turf/T = get_turf(src)
+	if(T && activity == TUMOR_ACTIVE)
+		for(var/t in RANGE_TURFS(12, T))
+			if(t && get_dist(t, T) == 12)
+				var/obj/effect/temp_visual/elite_tumor_wall/newwall
+				newwall = new /obj/effect/temp_visual/elite_tumor_wall(t, src)
+				newwall.activator = src.activator
+				newwall.ourelite = src.mychild
+		addtimer(CALLBACK(src, .proc/arena_checks), 50)
+		return
+
+/obj/structure/elite_tumor/proc/arena_checks()
+	INVOKE_ASYNC(src, .proc/arena_trap)  //Gets another arena trap queued up for when this one runs out.
+	INVOKE_ASYNC(src, .proc/border_check)  //Checks to see if our fighters got out of the arena somehow.
+	INVOKE_ASYNC(src, .proc/fighters_check)  //Checks to see if our fighters died.
+	return
+		
+/obj/effect/temp_visual/elite_tumor_wall
+	name = "magic wall"
+	icon = 'icons/turf/walls/hierophant_wall_temp.dmi'
+	icon_state = "wall"
+
+	duration = 50
+	smooth = SMOOTH_TRUE
+	layer = BELOW_MOB_LAYER
+	var/mob/living/carbon/human/activator = null
+	var/mob/living/simple_animal/hostile/asteroid/elite/ourelite = null
+	color = rgb(255,0,0)
+	light_range = MINIMUM_USEFUL_LIGHT_RANGE
+	light_color = LIGHT_COLOR_RED
+	
+/obj/effect/temp_visual/elite_tumor_wall/Initialize(mapload, new_caster)
+	. = ..()
+	queue_smooth_neighbors(src)
+	queue_smooth(src)
+
+/obj/effect/temp_visual/elite_tumor_wall/Destroy()
+	queue_smooth_neighbors(src)
+	return ..()
+
+/obj/effect/temp_visual/elite_tumor_wall/CanPass(atom/movable/mover, turf/target)
+	if(mover == ourelite || mover == activator)
+		return FALSE
+	else
+		return TRUE
+		
+/obj/structure/elite_tumor/proc/border_check()
```

Comment:
why do you jump from having code for the tumor back to the effect, then back to the tumor. Code relevant to the object in question should be grouped together.